src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { FC, useEffect, useMemo, useState } from 'react';
+import { Person } from '../types/Person';
+import { formatString } from '../servises/formatString';
+import debounce from 'lodash.debounce';
+
+type Props = {
+  people: Person[];
+  onSelect: (person: Person | null) => void;
+};
+
+export const Autocomplete: FC<Props> = ({ people, onSelect }) => {
+  const [query, setQuery] = useState('');
+  const [isDropdownVisible, setIsDropdownVisible] = useState(false);
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const applyQuery = useMemo(() => {
+    return debounce(setAppliedQuery, 300);
```

Comment:
The debounce delay is hardcoded to 300ms. According to the task requirements, the `delay` should be customizable via props with a default value of 300ms. Consider adding a `delay` prop to the component.